### DISCUSSION TITLE
Timer jitter analysis

### DISCUSSION THREAD

**Discussion by connorjburton** [oven-sh/bun#4294] [Show and tell]
As part of building a side project, I have performed some analysis of the Jitter between setting timers and their actual execution time.

It may be interesting to others to see how Bun compares to other JavaScript runtimes (and Rust as an added bonus). Happy for input on how to improve Bun's performance in this regard.

The code is in the repository for reference.

Here are some recent results to compare:

- Bun @ 100k timers https://github.com/connorjburton/replay-load-test/actions/runs/5912040615
- Node @ 100k timers https://github.com/connorjburton/replay-load-test/actions/runs/5912040612
- Node @ 300k timers https://github.com/connorjburton/replay-load-test/actions/runs/5931183869
- Deno @ 100k timers https://github.com/connorjburton/replay-load-test/actions/runs/5912040614
- Rust @ 750k timers https://github.com/connorjburton/replay-load-test/actions/runs/5918509394

**Comment by Jarred-Sumner** [comment]
yeah we need to redo our linux timers implementation, it's not efficient right now
